torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -307,23 +307,30 @@ def backend(self) -> str:
     def build_device_mesh(self) -> DeviceMesh:
         return DeviceMesh(self.device_type, list(range(self.world_size)))
 
-    def init_pg(self) -> None:
+    def init_pg(self, eager_init) -> None:
         if "nccl" in self.backend and torch.cuda.device_count() < self.world_size:
             sys.exit(TEST_SKIPS[f"multi-gpu-{self.world_size}"].exit_code)
 
         if self.backend not in ["nccl", "gloo", "mpi", "cpu:gloo,cuda:nccl"]:
             raise RuntimeError(f"Backend {self.backend} not supported!")
 
+        device = None
+        if "nccl" in self.backend:
+            # set device for nccl pg for collectives
+            torch.cuda.set_device(self.rank)
+            device = torch.device(f"{self.device_type}:{self.rank}") if eager_init else device
```

Comment:
nit: consider better portability here:
```
device = torch.device(f"{self.device_type}:{self.rank}")
if "nccl" in self.backend:
    ...
```
The `if eager_init` logic can be moved down to the `init_process_group` call.